### ISSUE TITLE
Improve handling of optional relations

### ISSUE THREAD

**Issue by alex3683** [alex3683/hal-http-client#18]
Some relations might be optional in a resource representation. Currently `follow` and `followAll` will fail if such a relation should be followed but doesn't exist. The only way to prevent this is by checking its existence by using `hal.canFollow`, but this is tedious and leads to ugly code.
It would be nice to simply pass an option to `follow` or `followAll` to indicate, that it's okay if a relation doesn't exist. In this case either `null` or an empty array should be passed to the on-Handler. I'm afraid I don't know yet which status code would be the most fitting. I'm torn between `200` and `204`.

**Comment by alex3683** [comment]
After thinking about this a bit longer I come to the conclusion, that perhaps something different than a real status code would be appropriate, e.g. `'norel'` to make explicit that the relation was missing. Guessing status codes feels wrong (and probably is).

**Comment by alex3683** [comment]
Implemented on `master` (`v0.8.0`).

## New Feature

The "virtual" status code `'norel'` can now be used to handle the case where a relation is missing when trying to follow it:
```js
hal.follow( peterResource, 'work' )
   .on( {
      '200': () => {
         console.log( 'Peter currently is employed' );
      },
      'norel': () => {
         console.log( 'Peter currently is unemployed' );
      }
   } );
```
